packages/babel-helpers/scripts/generate-helpers.js

```diff
@@ -177,7 +177,7 @@ const helpers: Record<string, Helper> = {
         mangle: {
           keep_fnames: mangleFns
             ? noMangleFns.length
-              ? new RegExp(noMangleFns.join("|"))
+              ? new RegExp("^" + noMangleFns.join("|") + "$")
```

Comment:
This should be with parentheses, otherwise ^/$ will only apply to the first/last of the options